stdlib/public/core/KeyPath.swift

```diff
@@ -2277,7 +2384,16 @@ internal func _tryToAppendKeyPaths<Result: AnyKeyPath>(
     }
     return _openExistential(rootValue, do: open2)
   }
-  return _openExistential(rootRoot, do: open)
+  var returnValue: AnyKeyPath = _openExistential(rootRoot, do: open)
+  _processOffsetForAppendedKeyPath(
+    appendedKeyPath: &returnValue,
+    root: root,
+    leaf: leaf
+  )
```

Comment:
I think it would be better to do this part inside of `open`, so that we don't lose the type information and have to re-cast the result in the code below. Alternatively, it should be impossible for `returnValue as? Result` to fail, so we can `unsafeDowncast` it instead.

Reply:
If I move this into `open3`, it can look like this:
```
var result:AnyKeyPath = _appendingKeyPaths(root: typedRoot, leaf: typedLeaf)
_processOffsetForAppendedKeyPath(
  appendedKeyPath: &result,
  root: root,
  leaf: leaf
)
return unsafeDowncast(result, to: Result.self)
```